src/awkward/_behavior.py

```diff
@@ -100,29 +100,32 @@ def find_ufunc_generic(
 
 
 def find_ufunc(behavior: Mapping | None, signature: tuple) -> UfuncLike | None:
-    if all(s is not None for s in signature):
-        behavior = overlay_behavior(behavior)
+    if any(s is None for s in signature):
+        return None
+
+    behavior = overlay_behavior(behavior)
 
-        # Special case all strings or hashable types.
+    # Try and fast-path the lookup
+    try:
+        return behavior[signature]
+    except KeyError:
```

Comment:
If a signature consists entirely of hashable things like the ufunc itself and strings matching a set of names in parameters (`__array__`, `__record__`, or `__list__`), then it gets fast-pathed and we get to use dict-lookup.

Good! :heavy_check_mark: